integration/tracing/tracing-opentelemetry/src/main/java/org/apache/cxf/tracing/opentelemetry/AbstractOpenTelemetryClientProvider.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.tracing.opentelemetry;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.apache.cxf.common.logging.LogUtils;
+import org.apache.cxf.phase.PhaseInterceptorChain;
+import org.apache.cxf.tracing.AbstractTracingProvider;
+import org.apache.cxf.tracing.opentelemetry.internal.TextMapInjectAdapter;
+
+public abstract class AbstractOpenTelemetryClientProvider extends AbstractTracingProvider {
+    protected static final Logger LOG = LogUtils.getL7dLogger(AbstractOpenTelemetryClientProvider.class);
+    protected static final String TRACE_SPAN = "org.apache.cxf.tracing.client.opentelemetry.span";
+
+    private final Tracer tracer;
+
+    public AbstractOpenTelemetryClientProvider(final Tracer tracer) {
+        this.tracer = tracer;
+    }
+
+    protected TraceScopeHolder<TraceScope> startTraceSpan(final Map<String, List<String>> requestHeaders,
+                                                          URI uri, String method) {
+        Context parentContext = Context.current();
+        Span activeSpan = tracer.spanBuilder(buildSpanDescription(uri.toString(), method))
+            .setParent(parentContext).setSpanKind(SpanKind.CLIENT)
+            .setAttribute(SemanticAttributes.HTTP_METHOD, method)
+            .setAttribute(SemanticAttributes.HTTP_URL, uri.toString())
+            // TODO: Enhance with semantics from request
+            .startSpan();
+        Scope scope = activeSpan.makeCurrent();
+
+        GlobalOpenTelemetry.getPropagators().getTextMapPropagator().inject(Context.current(), requestHeaders,
```

Comment:
The reference to statics (`GlobalOpenTelemetry`) should not be used, but it seems like in case of `OpenTelemetry`, having the instance of the `Tracer` is not sufficient. Should the `OpenTelemetry` + `instrumentationScopeName` be used instead?

Reply:
In the end, an OpenTelemetry object is passed around, and the tracer is retrieved only where it is used using a utility class.